proposals/apple-mdm/Fleet-MDM-Plan.md

```diff
@@ -0,0 +1,218 @@
+	- Creates `dep_names` entry with such token + depProfileUUID (as assigner_profile_uuid) (encryption of the token in MySQL will be a TODO.).
+	- Link such `dep_names` entry to the given `<ENROLLMENT_ID>`.
+- If all looks good, then it removes `fleet-mdm-apple-dep.key` from `~/.fleet/config`.
+
+#### Custom commands for DEP enrollments
+
+- List DEP devices: DEP devices of that enrollment and their status (using the DEP proxy API).
+	`fleetctl apple-mdm enrollments dep list <ENROLLMENT_ID>`
+- Sync DEP profiles: makes sure to set the enroll profile for new devices in a DEP enrollment. (Fleet would sync all DEP enrollments automatically every 5m.)
+	`fleetctl apple-mdm enrollments dep sync-profiles <ENROLLMENT_ID>`
+
+#### DEP syncer
+
+Fleet must start a "DEP syncer" routine for each entry in `apple_dep_enrollments` that has a non-NULL `dep_names` (period of 5m).
+
+### Hosts
+
+Initially, we want to account for hosts that are MDM enrolled but not in osquery.
+TODO(Lucas): We'll need to test whether our system and UI support a `hosts` entry w/ an empty `osquery_host_id`...
+
+### MDM commands
+
+These are one-off commands like "DeviceLock", "RestartDevice", etc.
+
+Fleetctl commands:
+- Send command:
+	`fleetctl apple-mdm commands send --target-hosts=1,2,3 --command=foo.plist`
+(fleetctl would be agnostic to contents of "foo.plist")
+- List commands (with their status):
+	`fleetctl apple-mdm commands list`
+
+### MDM profiles
+
+A profile is defined as:
+- ID
+- UUID
+- Name
+- Payload (raw XML plist)
+- Team (NULL for global)
+
+For MVP-dogfood, we will only have global profiles, and as such, they are applied to all MDM enrolled hosts.
+
+TODO(Lucas): Stuff to solve around this feature:
+So, we have a list of all the profiles to apply to devices. How does Fleet know which profiles are
+already deployed on a device and which aren't? (Are there any APIs to poll that work at scale?)
+
+Fleetctl commands (and APIs) to:
+- Create global profiles:
+	`fleetctl apple-mdm profiles create --name="Chrome_Enrollment" --payload=foo.xml`
+- List profile(s) (displays ID/UUID):
+	`fleetctl apple-mdm profiles list`
+- Delete profiles:
+	`fleetctl apple-mdm profiles delete`
+  TODO(Lucas): It will delete the profile from all hosts.
+
+### Application deployment
+
+#### Default applications to deploy
+
+In the current PoC, after MDM enrollment of a device, Fleet will auto-push the following commands:
+- "InstallApplication" to install Vanilla Munki
+- "InstallProfile" to configure Munki to connect to Fleet for software
+
+TODO(Lucas): Should Fleet check if the device has Munki installed and is configured correctly and then send commands when necessary?
+
+##### Osquery
+
+TODO(Lucas): There's the goal of pushing the orbit/osquery package to the instance right after MDM enroll (either via Munki or as a command).
+
+### Admin-defined applications
+
+TODO(Lucas): Munki...
```

Comment:
Does this entail a TUF server or something similar? Do I, as an IT admin, have security concerns about things being sent down?

Reply:
We will use [Munki](https://github.com/munki/munki) for software deployment. Which solve most problems of software deployment.

Munki has an agent that Fleet will auto-install on MDM enrolled hosts, and Fleet will be the server for such agent.

Reply:
We will add more details soon.